### PR TITLE
Issue 18598 invalid template npe

### PR DESCRIPTION
**Pull request by erickgonzalez**
Fix for NPE when PP a page when the template was already deleted(check if the template is not null), an Integration Test for this case was created.

Add annotations for the new filter methods.

Change calls of ES to DB calls when doing content management functions.

### CODE REVIEW

File: dotCMS/src/integration-test/java/com/dotcms/enterprise/publishing/remote/DependencyBundlerTest.java [hunk 1]
```diff
@@ -565,5 +568,34 @@ public void testGenerateBundleWithFilter_RelationshipsFalse_ChildcontentletChild
         Assert.assertTrue(listOfAssetsWithNewFilter.getRelationships().isEmpty());
     }
 
+    /**
+     * Method to test: {@link DependencyManager#setDependencies()}
+     * Given Scenario: If a page has a Template that was deleted, we should be able to generate the bundle without issues.
+     * ExpectedResult: The bundle should be generated without issues and the page added to the bundle.
+     */
+    @Test
+    public void Test_GenerateBundle_PageWithTemplateDeleted_PageSuccessfullyAddedToBundle() throws Exception {
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/enterprise/publishing/remote/DependencyBundlerTest.java:577]: Codacy Issue found: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5647397)

File: dotCMS/src/integration-test/java/com/dotcms/enterprise/publishing/remote/DependencyBundlerTest.java [hunk 2]
```diff
@@ -4,12 +4,14 @@
 
 import com.dotcms.IntegrationTestBase;
 import com.dotcms.LicenseTestUtil;
+import com.dotcms.content.elasticsearch.business.ESContentFactoryImpl;
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/enterprise/publishing/remote/DependencyBundlerTest.java:7]: Codacy Issue found: [Avoid unused imports such as 'com.dotcms.content.elasticsearch.business.ESContentFactoryImpl'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5647397)

File: dotCMS/src/integration-test/java/com/dotcms/enterprise/publishing/remote/DependencyBundlerTest.java [hunk 3]
```diff
@@ -4,12 +4,14 @@
...
 import com.dotcms.contenttype.model.field.Field;
 import com.dotcms.contenttype.model.field.FieldBuilder;
 import com.dotcms.contenttype.model.field.RelationshipField;
 import com.dotcms.contenttype.model.type.BaseContentType;
 import com.dotcms.contenttype.model.type.ContentType;
 import com.dotcms.datagen.ContainerDataGen;
+import com.dotcms.datagen.ContentletDataGen;
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/enterprise/publishing/remote/DependencyBundlerTest.java:14]: Codacy Issue found: [Avoid unused imports such as 'com.dotcms.datagen.ContentletDataGen'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5647397)

File: dotCMS/src/main/java/com/dotcms/publisher/util/DependencyManager.java [hunk 4]
```diff
@@ -816,15 +816,16 @@ private void setHTMLPagesDependencies(final Set<String> idsToWork,final Publishe
 
 						for (final MultiTree multiTree : treeList) {
 							final String contentIdentifier = multiTree.getChild();
-							final List<Contentlet> contentList = APILocator.getContentletAPI()
-									.search("+identifier:" + contentIdentifier, 0, 0, "moddate",
-											user, false);
+							final Identifier id = APILocator.getIdentifierAPI().find(contentIdentifier);
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/publisher/util/DependencyManager.java:819]: Codacy Issue found: [Avoid variables with short names like id](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5647397)

**Review by wezell — APPROVED** [review]
Nice test

**Review by nollymar — APPROVED** [review]

**Review by freddyucv — APPROVED** [review]

**Pull request merged**